code/modules/projectiles/gun.dm

```diff
@@ -150,6 +150,25 @@
 
 	if(!special_check(user))
 		return
+		
+	if(istype (src, /obj/item/weapon/gun/energy))
+		if(src.vent_stack)
+			return
+		if(src.heat_level = src.heat_cap)
+			user << "<span class='warning'>[src] feels hot in your hands!<span>"
+		if((src.heat_level - src.heat_cap) = 1)

```

Comment:
Should be elseif or you're running checks that you're already establishing as false earlier in the code.